### PR TITLE
Add tensor parallelism for RWKV

### PR DESCRIPTION
**Pull request by jahatef**

### CODE REVIEW

File: configs/neox_arguments.md [hunk 1]
```diff
@@ -843,6 +843,29 @@ Model Arguments
 
 
 
+- **dim_att**: int
```
Comment (Quentin-Anthony) [inline configs/neox_arguments.md:846]: we should either have unified args (across mamba, rwkv, transformers) for these, or prepend these args with whatever block type they're targeting (e.g. `rwkv_dim_att`).

File: configs/rwkv/430M.yml [hunk 2]
```diff
@@ -0,0 +1,103 @@
+{
+  # Parallelism is not yet supported for rwkv
+  "pipe_parallel_size": 1,
+  "model_parallel_size": 1,
+
+  "num_layers": 24,
+  "hidden_size": 1024,
+  "num_attention_heads": 16, # head_size = dim_att / num_attention_heads.
```
Comment (Quentin-Anthony) [inline configs/rwkv/430M.yml:8]: Similar comment here. Calling these attention heads is highly misleading.

File: megatron/model/rwkv/v6/rwkv.py [hunk 3]
```diff
@@ -7,7 +7,17 @@
...
-
+from megatron import mpu
+from megatron.mpu import gather_from_model_parallel_region, reduce_from_model_parallel_region, scatter_to_model_parallel_region
+try:
+    from fla.ops.rwkv6 import chunk_rwkv6
+    import einops
+except ModuleNotFoundError:
+    print(
+        "Unable to import RWKV FLA kernels. Install them from our requirements/requirements-rwkv.txt, \
+    or directly from https://github.com/sustcsonglin/flash-linear-attention.git, or use CUDA kernels."
```
Comment (Quentin-Anthony) [inline megatron/model/rwkv/v6/rwkv.py:18]: This last point `"or use CUDA kernels"` is confusing. Can you add a `"by doing xyz"` so that users know what you mean?

File: megatron/model/rwkv/v6/rwkv.py [hunk 4]
```diff
@@ -104,7 +126,7 @@ class RWKV_TimeMix(nn.Module):
     TODO: fix jit compiling.
```
Comment (Quentin-Anthony) [inline megatron/model/rwkv/v6/rwkv.py:126]: Is this based on the parser issue we discussed? I think it's worth testing just-jit and reordered jit and heuristics like I suggested before merging with this TODO

File: megatron/model/rwkv/v6/rwkv.py [hunk 5]
```diff
@@ -297,43 +388,45 @@ def __init__(self, neox_args, layer_number):
...
         self.ln1 = nn.LayerNorm(neox_args.hidden_size)
         self.ln2 = nn.LayerNorm(neox_args.hidden_size)
 
-        self.att = RWKV_TimeMix(neox_args, layer_number)
+        self.att = RWKV_TimeMix(neox_args, layer_number, init_method=init_method)
 
-        self.ffn = RWKV_ChannelMix(neox_args, layer_number)
+        self.ffn = ParallelRWKV_ChannelMix(neox_args, layer_number, init_method=init_method)
 
         if neox_args.attention_dropout > 0:
```
Comment (Quentin-Anthony) [inline megatron/model/rwkv/v6/rwkv.py:402]: another attention arg for rwkv. Can we decouple attn dropout from rwkv?

File: megatron/neox_arguments/arguments.py [hunk 6]
```diff
@@ -50,16 +50,19 @@
...
 GREEN = "\033[92m"
 RED = "\033[91m"
 YELLOW = "\033[93m"
-END = "\033[0m"
-SUCCESS = f"{GREEN} [SUCCESS] {END}"
-OKAY = f"{GREEN}[OKAY]{END}"
-WARNING = f"{YELLOW}[WARNING]{END}"
+
+### Formatted logging prefixes ###
+ERROR = f"{RED}[ERROR]{END} "
```
Comment (Quentin-Anthony) [inline megatron/neox_arguments/arguments.py:60]: I don't think we've properly merged this branch onto upstream `main`, since this is tracking as a change. Please do this.

File: megatron/neox_arguments/neox_args.py [hunk 7]
```diff
@@ -277,6 +277,11 @@ class NeoXArgsModel(NeoXArgsTemplate):
           }
     """
 
+    rwkv_fla: bool = False
```
Comment (Quentin-Anthony) [inline megatron/neox_arguments/neox_args.py:280]: regen `neox_arguments.md`, since this isn't showing up there.

File: megatron/training.py [hunk 8]
```diff
@@ -406,6 +406,9 @@ def get_batch(neox_args, data_iterator):
             datatype=datatype,
         )
     elif neox_args.train_impl == "kto":
+        assert (
```
Comment (Quentin-Anthony) [inline megatron/training.py:409]: I think these will also go away with a proper rebase onto latest `main`

File: configs/rwkv/430M.yml (hunk 2 above)
Comment (jahatef) [reply configs/rwkv/430M.yml:8]: I kind of disagree, as rwkv code generally references time mixing as `attention`, and the RWKV kernel is often called a type of "linear attention." But, I can add a bunch of configs to decouple rkwv and transformer config options, but this will just create a lot of config args that have essentially the same purpose in my opinion.

File: megatron/model/rwkv/v6/rwkv.py (hunk 3 above)
Comment (Quentin-Anthony) [reply megatron/model/rwkv/v6/rwkv.py:18]: reminder^
